File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> arr = new ArrayList<>();
        int n = speeds.length;
        Stack<Integer> st = new Stack<>();
        
        for(int i=0;i<n;i++) {
            int p = 100 - progresses[i], s = speeds[i];
            int work = (p + s - 1) / s;
            
            if(st.isEmpty() || st.firstElement() >= work) st.push(work);
            else {
                arr.add(st.size());
                st.clear();
                st.push(work);
            }
        }
        arr.add(st.size());
        
        int[] answer = new int[arr.size()];
        for(int i=0;i<arr.size();i++) {
            answer[i] = arr.get(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var answer = [Int]()
    var st = [Int]()
    for i in 0..<progresses.count {
        let need = (100 - progresses[i] + speeds[i] - 1) / speeds[i]
        if !st.isEmpty && st[0] < need {
            answer.append(st.count)
            st.removeAll()
        } 
        st.append(need)
    }
    if !st.isEmpty {answer.append(st.count)}
    return answer
}